src/app/model/Pawn.ts

```diff
@@ -0,0 +1,70 @@
+import { Board } from './Board';
+import { BOARDSIZE } from './Constances';
+import { Piece } from './Piece';
+import { PieceMovement } from './PieceMovement';
+import { Column, columns, Row, Side, Square } from './Types';
+
+export class Pawn extends Piece implements PieceMovement {
+  private nextRowDifference: number = 1;
+
+  constructor(id: string, side: Side) {
+    super(id, side);
+    if (this.side !== 'WHITE') this.nextRowDifference = -1;
+    console.log(this.nextRowDifference);
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [] as Square[];
```

Comment:
Tą całą metodę powinieneś dać radę skorcić do 1 linijki, nawet zwiększając czytelność :)
Pamiętajcie, że jak najmniej modyfikacji (const zamiast let), jak najmniej =, gdzie tylko można to stałe wartości.
A tutaj wartość jest od razu do zwrócenia :)

Reply:
Próbowałem na różne sposoby skrócić, aby na początku postawić pusta tablice a dodatkowe elementy w środku metody concat, ale nie potrafię.
Walczyłem z tym dziś z 20 minut

Reply:
Nie potrafię tego rozwikłać z zostawieniem w nawiasach tych metod które są w nawiasach!
A bardzo nie chciałbym ich wyjmować (ani jednej)
![image](https://user-images.githubusercontent.com/41823689/106336437-2e34f500-628f-11eb-9c54-ae8848420959.png)